emd_approx_relaxed.hh

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <cstdint>
+#include <algorithm>
+
+template <typename T>
+T emd_approx_relaxed (const T* w1, const T* w2, const T* dist, uint32_t size)
+{
+  static int idxes[1000*1000+7]; //to avoid any malloc in code
```

Comment:
I see what you mean. Can you please change the interface of this function to get `idxes` from the outside? E.g.

```c++
template <typename T>
 T emd_approx_relaxed (const T* w1, const T* w2, const T* dist, uint32_t size, std::shared_ptr<int32_t> cache)
```
The size and allocation of the cache will be a responsibility of the caller.